tests/Integration/Behaviour/Features/Context/Domain/AliasFeatureContext.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit\Framework\Assert;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Command\AddAliasCommand;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Query\GetAliasForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Alias\QueryResult\AliasForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Alias\ValueObject\AliasId;
+use Tests\Integration\Behaviour\Features\Context\SharedStorage;
+use Tests\Resources\DatabaseDump;
+
+class AliasFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add alias :reference with following information:
+     *
+     * @param string $reference
+     * @param TableNode $table
+     */
+    public function addAlias(string $reference, TableNode $table): void
+    {
+        $data = $table->getRowsHash();
+
+        $aliases = explode(',', $data['alias']);
+
+        /** @var AliasId[] $aliasIds */
+        $aliasIds = $this->getCommandBus()->handle(new AddAliasCommand(
+            $aliases,
+            $data['search']
+        ));
+
+        $this->getSharedStorage()->set($reference, $aliasIds);
```

Comment:
this is tricky, I wonder if we need to have some additional step to assert each created alias and save its reference (so that 1 reference = 1 id)? Similar to what we done with combinations (https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/CombinationListingFeatureContext.php#L248)? Because now multiple aliasIds are saved under single alias reference.
For now it works, but when we add scenario for editing to approve that one alias will update all related ones (and we will want to update different aliases having same searchTerm, then it will be a problem).
So after creation we would always have to add a check like
```
When I add alias with following information:
      | alias   | blouse, blus  |
      | search  | blouse        |
Then following aliases should exist:
    | reference | alias             | search_term | 
    | alias1       | bloose, blus | blouse        |
    | alias2       | bloose, blus | blouse        |
```
and inside the implementation we save those aliases ids to related references. Then once we need it in edition context we will be able to check that editing alias1 also affects alias2 and vice versa.

wdyt @jolelievre, am I overthinking? :smile:

Reply:
Yeah I think it's exactly like the combinations, so indeed the check after creation should use the `reference` column to assign a reference on each alias entity and making sure the reference is assigned to the right value (we shouldn't trust the order probably)

We should also use an `alias id` column that will be used to assert the ID value in following steps, so the step needs to check both `reference` (used to assign reference) and `alias id` (used to assert the ID value) at the same time